### PR TITLE
Fixes issue 756 geoJson namespace error

### PR DESCRIPTION
**Pull request by thomas-topway-it**
fixes https://github.com/ProfessionalWiki/Maps/issues/756
fixes 12 related errors running tests

the content of `fillParserOutput` has been fetched from Mediawiki 1.39.3 `JsonContentHandler` and `SemanticGeoJsonStore -> storeGeoJson` argument contains the encoded raw json. 

The following block seems to cause the issue

```
GeoJsonMapPageUi::forExistingPage( $content->beautifyJSON() )
			->addToOutput( OutputFacade::newFromParserOutput( $parserOutput ) );
```

### CODE REVIEW

File: src/GeoJsonPages/GeoJsonContentHandler.php [hunk 1]
```diff
@@ -34,19 +38,31 @@ protected function fillParserOutput(
...
+			$parserOutput->addModuleStyles( [ 'mediawiki.content.json' ] );
+		} else {
+			$parserOutput->setText( null );
 		}
 
-		GeoJsonMapPageUi::forExistingPage( $content->beautifyJSON() )
-			->addToOutput( OutputFacade::newFromParserOutput( $parserOutput ) );
+		// @FIXME alternatively decode $this->mText in GeoJsonLegacyContent
+		// to avoid decoding it again in SubObjectBuilder -> getSubObjectsFromGeoJson
+		$text = json_encode( $content->getData()->getValue() );
```
Comment (JeroenDeDauw) [inline src/GeoJsonPages/GeoJsonContentHandler.php:60]: Looks like this should be in the below if-block

File: src/GeoJsonPages/GeoJsonContentHandler.php [hunk 2]
```diff
@@ -34,19 +38,31 @@ protected function fillParserOutput(
...
+		// @see JsonContentHandler -> fillParserOutput
+		if ( $cpoParams->getGenerateHtml() ) {
+			if ( $content->isValid() ) {
+				$text = $content->getData()->getValue();
+				$parserOutput->setText( $content->rootValueTable( $content->getData()->getValue() ) );
+				
+			} else {
+				$error = wfMessage( 'invalid-json-data' )->parse();
+				$parserOutput->setText( $error );
+			}
```
Comment (JeroenDeDauw) [inline src/GeoJsonPages/GeoJsonContentHandler.php:51]: $parserOutput->setText( newMethod( args ) )

File: src/GeoJsonPages/GeoJsonContentHandler.php (hunk 1 above)
Comment (thomas-topway-it) [reply src/GeoJsonPages/GeoJsonContentHandler.php]: I have fixed this but not sure about the comment below

File: src/GeoJsonPages/GeoJsonContentHandler.php (hunk 2 above)
Comment (JeroenDeDauw) [reply src/GeoJsonPages/GeoJsonContentHandler.php:51]: @thomas-topway-it you added a bunch of random details that are now needed for compat reasons. It's best to encapsulate those in a dedicated function, to not pollute the main method.

Comment (JeroenDeDauw) [reply src/GeoJsonPages/GeoJsonContentHandler.php:51]: https://www.entropywins.wtf/blog/2018/10/30/readable-functions-do-one-thing/

Comment (thomas-topway-it) [reply src/GeoJsonPages/GeoJsonContentHandler.php]: fixed, now using the parent fillParserOutput and added additional condition

File: src/GeoJsonPages/GeoJsonContentHandler.php [hunk 3]
```diff
@@ -1,5 +1,9 @@
 <?php
 
+/**
+ * @reviewer thomas-topway-it for KM-A
+ */
```
Comment (JeroenDeDauw) [inline src/GeoJsonPages/GeoJsonContentHandler.php:5]: We do not add author or reviewer tags in this repo

File: src/GeoJsonPages/GeoJsonContentHandler.php [hunk 4]
```diff
@@ -35,18 +39,20 @@ protected function fillParserOutput(
 	) {
 		'@phan-var GeoJsonContent $content';
 
-		if ( !$cpoParams->getGenerateHtml() || !$content->isValid() ) {
-			$parserOutput->setText( '' );
-			return;
-		}
+		parent::fillParserOutput( $content, $cpoParams, $parserOutput );
```
Comment (JeroenDeDauw) [inline src/GeoJsonPages/GeoJsonContentHandler.php:42]: Why did you add this call?

Comment (thomas-topway-it) [reply src/GeoJsonPages/GeoJsonContentHandler.php:42]: this is to set the correct parserOutput text by calling the same method in the parent class (JsonContentHandler)
